Sources/Fuzzilli/FuzzIL/JSTyper.swift

```diff
@@ -170,7 +170,36 @@ public struct JSTyper: Analyzer {
     /// Attempts to infer the parameter types of the given subroutine definition.
     /// If parameter types have been added for this function, they are returned, otherwise generic parameter types (i.e. .anything parameters) for the parameters specified in the operation are generated.
     private func inferSubroutineParameterList(of op: BeginAnySubroutine, at index: Int) -> ParameterList {
-        return signatures[index] ?? ParameterList(numParameters: op.parameters.count, hasRestParam: op.parameters.hasRestParameter)
+        if let signature = signatures[index] {
+            return signature
+        } else {
+            var parameterList = ParameterList()
+            let patterns = op.parameters.patterns
+            let hasRestParam = op.parameters.hasRestParameter
+
+            for (i, pattern) in patterns.enumerated() {
+                let ilType = inferParameterType(from: pattern)
+                let parameter: Parameter
+                if hasRestParam && i == patterns.count - 1 {
+                    parameter = .rest(ilType)
+                } else {
+                    parameter = .plain(ilType)
+                }
+                parameterList.append(parameter)
+            }
+            return parameterList
+        }
+    }
+
+    private func inferParameterType(from pattern: ParameterPattern) -> ILType {
+        switch pattern {
+        case .identifier:
+            return .anything
+        case .array:
+            return .iterable
+        case .object:
+            return .object()
```

Comment:
Maybe leave a TODO here that we could be more precise (IIUC), e.g. by adding properties with the parameter names (e.g. `.object(withProperties: ["a"])`, etc.)

Reply:
Good idea. I am wondering if being precise here is something that we must do. What happens if we do not specify details about an object ILType?

Reply:
Ah ok that answers the question